spec/parse_date/int_from_string_spec.rb

```diff
@@ -0,0 +1,578 @@
+    '-999' => '-001',
+    '-914' => '-086',
+    '-18' => '-982',
+    '-1' => '-999',
+    '0' => '0000',
+    '5' => '0005',
+    '33' => '0033',
+    '945' => '0945'
+  }
+  bc_dates_to_int = {
+    '801 B.C.' => -801,
+    '800 B.C.' => -800,
+    '750 B.C.' => -750,
+    '700 B.C.' => -700,
+    '699 B.C.' => -699,
+    '75 B.C.' => -75,
+    '8 B.C.' => -8
+  }
+
+  context '#year_int_from_date_str' do
+    single_year
+      .merge(specific_month)
+      .merge(specific_day)
+      .merge(specific_day_2_digit_year)
+      .merge(bc_dates_to_int)
+      .merge(specific_day_ruby_parse_fail)
+      .merge(brackets_in_middle_of_year)
+      .merge(invalid_but_can_get_year).each do |example, expected|
+      it "#{expected} for single value #{example}" do
+        expect(DummyClass.new.year_int_from_date_str(example)).to eq expected.to_i
+      end
+    end
+
+    multiple_years
+      .merge(multiple_years_4_digits_once)
+      .merge(decade_only)
+      .merge(decade_only_4_digits).each do |example, expected|
+      it "#{expected.first} for multi-value #{example}" do
+        expect(DummyClass.new.year_int_from_date_str(example)).to eq expected.first.to_i
+      end
+    end
+
+    century_only.keys.each do |example|
+      it "1700 from #{example}" do
+        expect(DummyClass.new.year_int_from_date_str(example)).to eq 1700
+      end
+    end
+
+    early_numeric_dates.each do |example, _expected|
+      it "#{example} for #{example}" do
+        expect(DummyClass.new.year_int_from_date_str(example)).to eq example.to_i
+      end
+    end
+
+    it 'nil for -1666' do
+      skip('code broken for -yyyy dates but no existing data for this yet')
+      expect(DummyClass.new.year_int_from_date_str('-1666')).to eq nil
+    end
+    it '-1666 for 1666 B.C.' do
+      expect(DummyClass.new.year_int_from_date_str('1666 B.C.')).to eq(-1666)
+    end
+
+    [ # bad dates
+      '9999',
+      '2035',
+      '0000-00-00',
+      'uuuu',
+      '1uuu'
+    ].each do |example|
+      it "nil for #{example}" do
+        expect(DummyClass.new.year_int_from_date_str(example)).to eq nil
+      end
+    end
+  end
+
+  context '#year_int_valid?' do
+    { # example int as key, expected result as value
+      -1666 => false,
+      -999 => true,
+      -35 => true,
+      -3 => true,
+      0 => true,
+      5 => true,
+      33 => true,
+      150 => true,
+      (Date.today.year + 1) => true, # current year + 1
+      (Date.today.year + 2) => false, # current year + 2
+      9999 => false,
+      '165x' => false,
+      '198-' => false,
+      'random text' => false,
+      nil => false
+    }.each do |example, expected|
+      it "#{expected} for #{example}" do
+        expect(DummyClass.new.year_int_valid?(example)).to eq expected
+      end
+    end
+    it 'true for 0000' do
+      expect(DummyClass.new.year_int_valid?(0000)).to eq true
+    end
+  end
+
+  context '#sortable_year_for_yyyy' do
```

Comment:
why bother testing this private method if all it's public usages are already tested via the `year_int_from_date_str` tests?

Reply:
direct port ... easier to just port over existing tests.  Plus, there's a lot of edge case functionality to test at individual private method level.